### PR TITLE
Update `JSOCClient.fetch` to accomodate breaking changes in `drms`

### PR DESCRIPTION
**Pull request by nabobalis**
Also pins `drms` to `>=0.7.1`.

### CODE REVIEW

**Comment by nabobalis** [comment]
Need to double check the online build passes first before merge.

**Comment by nabobalis** [comment]
```
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/database/tests/test_tables.py::test_entry_from_qr_block_with_missing_physobs
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/tests/test_fido.py::test_path
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/vso/tests/test_vso.py::test_path
FAILED ../../docs/tutorial/acquiring_data/index.rst::index.rst
```

These looks like VSO <-> JSOC issues

File: sunpy/net/jsoc/jsoc.py
Comment (Cadair) [inline sunpy/net/jsoc/jsoc.py:441]: So now we don't get messages with the export status any more?

Comment (nabobalis) [reply sunpy/net/jsoc/jsoc.py:441]: You get via logging.info, no keyword anymore.

**Review by dstansby — CHANGES_REQUESTED** [review]
This should have a changelog, explaining the difference in logging/progress bar behaviour with an upgrade to `drms` 0.7 (since users of sunpy won't neccessarily know about drms, we should tell them if the user experience of downloading from JSOC is changed)

File: sunpy/net/jsoc/jsoc.py
Comment (wtbarnes) [reply sunpy/net/jsoc/jsoc.py:441]: Seeems sensible to me and is consistent with how we control output in the rest of the package.

File: changelog/7307.trivial.rst
Comment (dstansby) [inline changelog/7307.trivial.rst:1]: Does this mean no progress bar any more? Does the progress show by default? How do I turn on and off the progress now compared to before? (are all questions that I would probably ask as a user looking at this change)

Comment (dstansby) [inline changelog/7307.trivial.rst:1]: Does this mean no progress bar any more? Does the progress show by default? How do I turn on and off the progress now compared to before? (are all questions that I would probably ask as a user looking at this change)

Comment (wtbarnes) [reply changelog/7307.trivial.rst:1]: All fair questions. Doing some manual testing, I find that the only way I'm able to control the verbosity of the progress bar is still through the `progress`keyword argument to `JSOCClient.fetch`. For example, the following code snippet still produces a progress bar, even though I've set the log level to "warn" (i.e. exclude "info"),

```python
import sunpy
sunpy.log.setLevel('WARN')

import astropy.units as u
from sunpy.net import jsoc
from sunpy.net import attrs as a
client = jsoc.JSOCClient()
response = client.search(a.Time('2014-01-01T00:00:00', '2014-01-01T00:00:10'),
                         a.jsoc.Series('aia.lev1_euv_12s'),
                         a.jsoc.PrimeKey('WAVELNTH','171'))
file = client.fetch(response, progress=True)
```
Given this, I'm confused as to what has actually changed on the `drms` side. Maybe I'm misunderstanding something here.

Comment (wtbarnes) [reply changelog/7307.trivial.rst:1]: Resolving this one as the same comment seems to have been posted twice.

Comment (wtbarnes) [reply changelog/7307.trivial.rst:1]: If we cannot figure this out in the next few days, we can close this and just carry on with the `<0.7` pin that we already have. I don't really understand the point of `drms` v0.7 anyway...

Comment (nabobalis) [reply changelog/7307.trivial.rst:1]: Drms with the progress keyword should print:

```
            if verbose:
                print(f"Downloading file {int(i + 1)} of {int(ndata)}...")
                print(f"    record: {di.record}")
                print(f"  filename: {di.filename}")
                print(f"  -> {os.path.relpath(fpath_new)}")

```
These are now logging.info

**Comment by nabobalis** [comment]
I updated the changelog to say what messages were affected. 

Triggering them isn't common, and it won't happen if the request is already complete.

File: setup.cfg
Comment (nabobalis) [inline setup.cfg:72]:
```suggestion
  drms>=0.7.1
```

File: sunpy/net/jsoc/jsoc.py
Comment (nabobalis) [inline sunpy/net/jsoc/jsoc.py:551]: I changed this as we should really just be dumping to the logger instead of a print. But maybe others like it this way?

Comment (wtbarnes) [inline sunpy/net/jsoc/jsoc.py:551]:
```suggestion
                log.info(f"{len(urls)} URLs found for download. Full request totaling {request._d['size']}MB")
```

Comment (wtbarnes) [reply sunpy/net/jsoc/jsoc.py:551]: I'm in favor of all of these messages being handled by the logger as it gives the user a standardized way to control the output.

**Review by wtbarnes — APPROVED** [review]

File: changelog/7307.trivial.rst
Comment (dstansby) [inline changelog/7307.trivial.rst:11]: It's not clear if these messages are now shown by default or not? And if not, could you add a quick explanation (with code) on how to turn them back on? (or say that it's not possible to see them any more if it isn't possible)

Comment (nabobalis) [reply changelog/7307.trivial.rst:11]: Added more text and code example. Hopefully that is good enough.

Comment (dstansby) [reply changelog/7307.trivial.rst:11]: 👍 liking the code example

File: changelog/7307.trivial.rst
Comment (dstansby) [inline changelog/7307.trivial.rst:3]:
```suggestion
As a result, we increased the minimum required version of `drms` to 0.7.1.
```

**Review by dstansby — APPROVED** [review]
LGTM, modulo fixing the verison number in the changelog.

**Pull request merged**
